test/cmdlineTests.sh

```diff
@@ -300,20 +302,24 @@ printTask "Running general commandline tests..."
         # Use printf to get rid of the trailing newline
         inputFile=$(printf "%s" "${inputFiles}")
 
-        # If no files specified, assume input.sol as the default
-        if [ -z "${inputFile}" ]; then
-            inputFile="${tdir}/input.sol"
-        fi
-
         if [ "${inputFile}" = "${tdir}/input.json" ]
         then
+            ! [ -e "${tdir}/stdin" ] || { printError "Found a file called 'stdin' but redirecting standard input in JSON mode is not allowed."; exit 1; }
+
             stdin="${inputFile}"
             inputFile=""
             stdout="$(cat "${tdir}/output.json" 2>/dev/null || true)"
             stdoutExpectationFile="${tdir}/output.json"
             command_args="--standard-json "$(cat "${tdir}/args" 2>/dev/null || true)
         else
-            stdin=""
+            if [ -e "${tdir}/stdin" ]
+            then
+                stdin="${tdir}/stdin"
+                [ -f "${tdir}/stdin" ] || { printError "'stdin' is not a regular file."; exit 1; }
+            else
+                stdin=""
```

Comment:
What happens if this `else` branch is taken? Is an error printed later on since `stdin` does not exist?

Reply:
No, it was the default behavior before my change. The stdin is empty in that case but `solc` still gets the name of the input file as one of its arguments.